### ISSUE TITLE
MACD in Ranging Markets

### ISSUE THREAD

**Issue by kan-il** [binary-com/binary-bot#1862]
Hi, I'm trying to create a bot with MACD, which works well in a trending market but opens losing trades when the market comes to a ranging/swinging state. Please tell me another indicator that I can use along with MACD, so that the bot won't go into trades in a swinging market or any other way that I can avoid the bot making losses in swinging markets. I don't need many trades, less than 5 trades per day is enough.

Thanks in advance...

**Comment by binary-ex-machina** [comment]
It's impossible to avoid sideways market completely. No matter the indicator, they will always enter on sideways, sooner or later

**Comment by hotmatrixx** [comment]
> 
> 
> Hi, I'm trying to create a bot with MACD, which works well in a trending market but opens losing trades when the market comes to a ranging/swinging state. Please tell me another indicator that I can use along with MACD, so that the bot won't go into trades in a swinging market or any other way that I can avoid the bot making losses in swinging markets. I don't need many trades, less than 5 trades per day is enough.
> 
> Thanks in advance...

technically, the MACD Histogram is a measure of market volitility. if that's above the line and rising, its trending up. if it's below the line and falling, it's trending down. anything else is probably sideways.

additionally, the Bollinger Bands are a measure of actual volitility and can show a (lagging) strength of a direction, when the upper and lower bands are wide apart, and a sideways market when the bands are near.

You will have to discern your own values for these indicators, and what your thresholds are for buying. as you can see, from the picture combining these is a pretty good measure of market trend direction.

this info is pretty readily available on the net, but most people wouldnt combine these indicators like this, as they are 'traditional' traders, and look for longer trends than what binary traders do. 

I did not go looking for a part of my graph to cherry pick a result to show you how it works, that's literally just what was on my screen when I decided to answer your question. It does not show all of my indicators, however it does explain my answer pretty clearly.
![image](https://user-images.githubusercontent.com/49715381/63573069-7fd84b00-c5d8-11e9-9d03-cdd3e50b546c.png)

**Comment by kan-il** [comment]
@binary-ex-machina @hotmatrixx really appreciate you guys for giving me your support. @hotmatrixx u have put some great effort to explain things to me, thanks a lot. While I understood what you explained above, it is difficult for me to code this into a bot since I'm new to this, therefore it would be great if both of you can help me by telling another indicator that I can use along with MACD to get better results. Thanks a lot guys <3

**Comment by binary-ex-machina** [comment]
@kan-il i can say add a cross signal from EMA and SMA. I can say to add a Stochastic signals cross as well. I can say to add a RSI indicator as filter. I can say to add another MACD with different settings. 

Like you can see you can add as many filters you want, all of them can end with the same win rate / profit. In case you don't know what to add, it's because the strategy is not even formed and you don't have a base indicator yet. MACD is good in many points, but as any other indicator, it was created for trades without expiration (forex, stocks and so on). There is no indicator created for binary options and it will never exist, because no indicator can predict the future, and binary options have expiration date, so anything can happen (50/50 chance of win/lose)

In case you struggling with the indicator, change it to another. Put in a graph, check the signals and see how profitable it is

**Comment by hotmatrixx** [comment]
> 
> 
> @binary-ex-machina @hotmatrixx really appreciate you guys for giving me your support. @hotmatrixx u have put some great effort to explain things to me, thanks a lot. While I understood what you explained above, it is difficult for me to code this into a bot since I'm new to this, therefore it would be great if both of you can help me by telling another indicator that I can use along with MACD to get better results. Thanks a lot guys <3

I'll put a simple MACD bot together for you so that you can see how they are built.
I don't mind helping people if they want to learn.

**Comment by hotmatrixx** [comment]
[simplified MACD and simplified Martingale.zip](https://github.com/binary-com/binary-bot/files/3536518/simplified.MACD.and.simplified.Martingale.zip)

now this MACD looks big and scary, but it's not.
it looks at the latest and previous candles. 
if the macd line is bigger than the itself, and it's bigger than the signal line, and the histo is going up, it purchases a rise.
if instead everything is pointing down, it purchases a fall
I have a simple notify in there to let me know that it's doing something even when it looks like it's doing nothing

in block 4 I added a super simple martingale.
this bot actually works pretty well, if you have 10k in the bank and dont mind it risking up to $2000 or more on a single trade. it's not an ideal strategy on it's own. its not the tidiest way to code it. its just a good way to get you familiar with MACD.

I also added a couple of basic functions to the Runonce so that you get an idea of how they can work.
